### PR TITLE
MetaEval changes from paper

### PR DESCRIPTION
**Pull request by krzentner**
Included is this is tests and everything needed to use the MetaEvaluator
as a field on an algorithm.

### CODE REVIEW

File: src/garage/_functions.py [hunk 1]
```diff
@@ -34,10 +34,14 @@ def log_multitask_performance(itr, batch, discount, name_map=None):
             task_name = trajectory.env_infos['task_name'][0]
         except KeyError:
             try:
+                task_id = None
```
Comment (krzentner) [inline src/garage/_functions.py]: I think Keren has an alternative change to this logic, which looks a little better.
I just needed the `MetaEvaluator` to work on environments where all tasks are unnamed / continuous.

Comment (krzentner) [reply src/garage/_functions.py]: Yeah, when Keren merges PR #1214 I'll remove the logic here.

**Comment by ahtsan** [comment]
This will close #1223

File: src/garage/experiment/meta_evaluator.py [hunk 2]
```diff
@@ -22,40 +24,61 @@ class MetaEvaluator:
...
         adapted_trajectories = []
+        logger.log('Sampling for adapation and meta-testing...')
+        if self._test_sampler is None:
+            self._test_sampler = LocalSampler.from_worker_factory(
+                WorkerFactory(seed=get_seed(),
+                              max_path_length=self._max_path_length,
+                              n_workers=1,
+                              worker_class=DefaultWorker),
+                agents=algo.get_exploration_policy(),
+                envs=self._test_task_sampler.sample(1))
```
Comment (ahtsan) [inline src/garage/experiment/meta_evaluator.py]: Should it be 
```
...
envs=self._test_task_sampler.sample(self._n_test_tasks))
```
?

Comment (krzentner) [reply src/garage/experiment/meta_evaluator.py]: Nope. The meta evaluator goes through one task at a time. See the next line of code.

Comment (ahtsan) [reply src/garage/experiment/meta_evaluator.py]: Since in `meta_squash` branch it is, just want to make sure it's the correct implementation here

Comment (krzentner) [reply src/garage/experiment/meta_evaluator.py]: Hmm. I'm fairly certain that part of the `meta_squash` branch is wrong, but thanks for checking.

**Pull request merged**